lib/ansible/modules/system/ufw.py

```diff
@@ -336,14 +440,34 @@ def ufw_version():
             if (ufw_major == 0 and ufw_minor >= 35) or ufw_major > 0:
                 cmd.append([params['comment'], "comment '%s'" % params['comment']])
 
-            execute(cmd)
+            rules_dry = execute(cmd)
 
-    # Get the new state
-    (_, post_state, _) = module.run_command(ufw_bin + ' status verbose')
-    (_, post_rules, _) = module.run_command("grep '^### tuple' /lib/ufw/user.rules /lib/ufw/user6.rules /etc/ufw/user.rules /etc/ufw/user6.rules")
-    changed = (pre_state != post_state) or (pre_rules != post_rules)
+            if module.check_mode:
+
+                nb_skipping_line = len(filter_line_that_contains("Skipping", rules_dry))
 
-    return module.exit_json(changed=changed, commands=cmds, msg=post_state.rstrip())
+                if not (nb_skipping_line > 0 and nb_skipping_line == len(rules_dry.splitlines(True))):
+
+                    rules_dry = filter_line_that_not_start_with("### tuple", rules_dry)
+                    # ufw dry-run doesn't send all rules so have to compare ipv4 or ipv6 rules
+                    if is_match_ipv4(params['from_ip']) or is_match_ipv4(params['to_ip']):
```

Comment:
Sorry, I think I got this wrong. Using a match function here is a bad idea, since `from_ip` and `to_ip` can also be subnets, not only addresses, and your regexp does not match subnets in the format `<address>/<bitsize>`. So we either have to use contains here, or we have to extend the regexps. Adding something like `(?:/[0-9]+)?` to them should do the job, though.

Reply:
I think that we can consider after ufw execution with a code return equal to 0 mean that the ip address is valid. 

So we can perform simple seek of the IP address pattern check what IP format the rules is targeting (ipv4, ipv6 or both).

Reply:
Yep, that's true. Either use `search` instead of `match`, or extend the pattern (or both).

Reply:
In fact using the match function correspond to ^PATTERN so the current code is working for me in this configuration. As you can see in the unit test I've added.

[1] https://docs.python.org/2/library/re.html#search-vs-match

Reply:
Interestingly, I always thought that `match()` corresponds to `^PATTERN$` and not just `^PATTERN`. In that case, everything's fine :)

(Except that the function's name could be more precise, `is_match_ipv4()` sounds to me that it only matches a IPv4 address with nothing else after it. But that's a matter of taste, I guess.)

Reply:
with unit test, renaming is not a problem :)